orangecontrib/network/community.py

```diff
@@ -181,7 +181,14 @@ def next_label(neighbors):
             if stop:
                 break
 
+    labels = remap_labels(labels)
+
     if results2items:
-        add_results_to_items(G, lblhistory)
+        add_results_to_items(G, labels)
 
     return labels
+
+def remap_labels(labels):
+    unique_labels = sorted(set(labels.values()))
+    dict_labels = dict(zip(unique_labels, range(len(unique_labels))))
```

Comment:
In general, `zip(range(len(s)), s)` can be replaced with `enumerate(s)`. You have the opposite order, so you can't just replace `zip` with `enumerate`. What I'd probably do here is `{label:i for i, label in enumerate(unique_labels)}`. Fewer parentheses. :)

I prefer comprehension since it *describes what is in the dictionary*, it shows me the key-value pairs. `dict` constructor separates them - it gets two lists that then get paired up. It's less obvious, at least to me.